tests/fuzz/s2n_equivalent_cert_pem_parsing_fuzz_test.c

```diff
@@ -0,0 +1,111 @@
+/*
+ * Copyright 2017 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+#include <errno.h>
+#include <fcntl.h>
+#include <stddef.h>
+#include <stdint.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <unistd.h>
+
+#include <openssl/ssl.h>
+#include <openssl/conf.h>
+#include <openssl/rand.h>
+#include <openssl/err.h>
+#include <openssl/x509.h>
+#include <openssl/dh.h>
+#include <openssl/ec.h>
+
+#include "api/s2n.h"
+#include "stuffer/s2n_stuffer.h"
+#include "tls/s2n_config.h"
+#include "utils/s2n_blob.h"
+#include "utils/s2n_mem.h"
+#include "utils/s2n_safety.h"
+#include "s2n_test.h"
+
+static void s2n_fuzz_atexit()
+{
+    s2n_cleanup();
+    ERR_free_strings();
+    CRYPTO_cleanup_all_ex_data();
+    CONF_modules_free();
+    ERR_clear_error();
+}
+
+int LLVMFuzzerInitialize(const uint8_t *buf, size_t len)
+{
+#ifdef S2N_TEST_IN_FIPS_MODE
+    S2N_TEST_ENTER_FIPS_MODE();
+#endif
+
+    GUARD(s2n_init());
+    GUARD(atexit(s2n_fuzz_atexit));
+    GUARD(setenv("S2N_ENABLE_CLIENT_MODE", "1", 0));
+
+    return 0;
+}
+
+int LLVMFuzzerTestOneInput(const uint8_t *buf, size_t len)
+{
+    struct s2n_stuffer in;
+    GUARD(s2n_stuffer_alloc(&in, len + 1));
+    GUARD(s2n_stuffer_write_bytes(&in, buf, len));
+    in.blob.data[len] = 0;
+
+    uint8_t parsable_by_openssl = 0;
+
+    /* Try parsing Cert PEM with OpenSSL */
+    BIO *membio = BIO_new_mem_buf((void *) in.blob.data, len);
+    if (membio != NULL) {
+        X509 *cert = PEM_read_bio_X509(membio, NULL, 0, NULL);
```

Comment:
To make this match our s2n implementation(which takes the entire chain). I think this needs to loop over `PEM_read_bio_X509` until it returns `NULL`.

Perhaps we can also compare the # of certs Openssl parsed vs # of certs s2n parsed? For example, there may be chain inputs which s2n believes contains 2 certs but Openssl is able to parse 3.